File: snippets/typescript/update_vectors.ts
import { QdrantClient } from "@qdrant/js-client-rest";

const client = new QdrantClient({ host: "localhost", port: 6333 });

client.updateVectors("global_patient_data", {
  points: [
    {
      id: 1,
      vector: {
        HealthVec: [0.1, 0.2, 0.3],
      },
    },
    {
      id: 2,
      vector: {
        DemoMedVec: [0.9, 0.8, 0.7],
      },
    },
  ],
});

File: snippets/go/update_vectors.go
package client

import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

func updateVectors() {
	client, err := qdrant.NewClient(&qdrant.Config{
		Host: "localhost",
		Port: 6334,
	})
	if err != nil {
		panic(err)
	}

	_, err = client.UpdateVectors(context.Background(), &qdrant.UpdatePointVectors{
		CollectionName: "{collection_name}",
		Points: []*qdrant.PointVectors{
			{
				Id: qdrant.NewIDNum(1),
				Vectors: qdrant.NewVectorsMap(map[string]*qdrant.Vector{
					"image": qdrant.NewVector(0.1, 0.2, 0.3, 0.4),
				}),
			},
			{
				Id: qdrant.NewIDNum(2),
				Vectors: qdrant.NewVectorsMap(map[string]*qdrant.Vector{
					"text": qdrant.NewVector(0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2),
				}),
			},
		},
	})
	if err != nil {
		panic(err)
	}
}
